scio-core/src/main/scala/com/spotify/scio/testing/TestDataManager.scala

```diff
@@ -144,17 +148,20 @@ private[scio] object TestDataManager {
     testId: String,
     ins: Map[String, JobInputSource[_]],
     outs: Map[String, SCollection[_] => Any],
-    dcs: Map[DistCacheIO[_], _]
+    dcs: Map[DistCacheIO[_], _],
+    xformOverrides: Set[PTransformOverride]
   ): Unit = {
     inputs += (testId -> new TestInput(ins))
     outputs += (testId -> new TestOutput(outs))
```

Comment:
Theoretically, `ins` and `outs` could also be implemented in terms of transform overrides rather than the ugly ugly ugly implementation we have now. Is that something you've tried @kellen ?

Reply:
No, this is my first attempt at poking at this code, but that does seem possible to do in follow-up work.